icu4c/source/common/dictbe.cpp

```diff
@@ -1053,6 +1061,9 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
     // Korean dictionary only includes Hangul syllables
     fHangulWordSet.applyPattern(UnicodeString(u"[\\uac00-\\ud7a3]"), status);
     fHangulWordSet.compact();
+    fNumberAndOpenPunctuationSet.applyPattern(UnicodeString(u"[[:Nd:][:Pi:][:Ps:]]"), status);
```

Comment:
This should be named fNumberOrOpenPunctuationSet not fNumberAndOpenPunctuationSet since it match an Unicode is a Number OR it is an Open Punctuation but not require it to be BOTH (which is actually impossible)